File: p62048.cpp
#include<bits/stdc++.h>
using namespace std;

int gcd(int a, int b){
    int tmp;
    while(b != 0){
        tmp = a % b;
        a = b;
        b = tmp;
    }
    return a;
}

long long solution(int w,int h) {
    int g = gcd(w, h);
    long long answer = ((long)w * h) - (((long)w / g) + ((long)h / g) - 1) * g;

    return answer;
}

File: p62048.java
import java.math.BigInteger;

public class p62048 {
    public static long solution(int w, int h) {
        long gcd = BigInteger.valueOf(w).gcd(BigInteger.valueOf(h)).longValue();
        long answer = ((long)w * (long)h) - ((((long)w / gcd) + ((long)h / gcd) - (long)1) * gcd);
        return answer;
    }

    public static void main(String[] args){
        System.out.println(solution(8, 12));
        System.out.println(solution(6, 9));
    }
}
